frontend/lib/resolution/InitResolver.cpp

```diff
@@ -434,7 +434,9 @@ static const ArrayType* arrayTypeFromSubsHelper(
                                    QualifiedType(QualifiedType::TYPE, domain),
                                    eltType);
   } else if (instanceBct->id().symbolPath() ==
-             "DefaultAssociative.DefaultAssociativeArr") {
+                 "DefaultAssociative.DefaultAssociativeArr" ||
+             instanceBct->id().symbolPath() ==
+                 "ArrayViewReindex.ArrayViewReindexArr") {
```

Comment:
I suspect this means your PR isn't just about adding tests, so I would change the title appropriately